src/config/mongoose_config_parser.erl

```diff
@@ -200,3 +201,58 @@ add_dep_modules_opt({local_config, {modules, Host}, Modules}) ->
     {local_config, {modules, Host}, Modules2};
 add_dep_modules_opt(Other) ->
     Other.
+
+%% local functions
+-ifdef(TEST).
+halt_with_msg(ExitMsg, Errors) ->
+    config_error(ExitMsg, Errors).
+-else.
+halt_with_msg(ExitMsg, Errors) ->
+    [?LOG_ERROR(Error) || Error <- Errors],
+    mongoose_config_utils:exit_or_halt(ExitMsg).
+-endif.
+
+config_error(ErrorMsg, Errors) ->
+    error({config_error, ErrorMsg, Errors}).
+
+-spec check_dynamic_domains_support(mongoose_config_parser:state()) -> any().
+check_dynamic_domains_support(State) ->
+    %% we must check that all the the modules configured for
+    %% the pure host types support dynamic domains feature
+    Config = get_opts(State),
+    HostTypes = state_to_host_types(State),
+    FoldFN = fun
+                 (#local_config{key = {modules, H}, value = Modules}, Acc) ->
+                     case lists:member(H, HostTypes) of
+                         false -> Acc;
+                         true ->
+                             BadModules = check_modules_for_host_type(Modules),
+                             Acc ++ invalid_modules_for_host_type(H, BadModules)
+                     end;
+                 (_, Acc) -> Acc
+             end,
```

Comment:
Could you use a named function here to improve debugging?

Reply:
It must be an anonymous function because of the closure (HostTypes variable).

so we have 2 options here:
* I can either create a function accepts HostTypes as input parameter and then generates FoldN function.
* Or move current FoldN implementation inside some named function that accepts 3 parameters (HostTypes, Element, Acc) and make a call of that function from the anonymous FoldN function.

I think only the second one can improve debugging a bit, but still want to clarify this. Which one were you having in mind?

Reply:
I meant the second option (replace the fun with a 3-argument function) that would also make the code structure a bit more flat.